prow/cmd/sinker/main.go

```diff
@@ -297,7 +297,18 @@ func (c *controller) clean() {
 
 	// Clean up old prow jobs first.
 	prowJobs := &prowapi.ProwJobList{}
-	if err := c.prowJobClient.List(c.ctx, prowJobs, ctrlruntimeclient.InNamespace(c.config().ProwJobNamespace)); err != nil {
+	timeoutSecs := (*int64)(nil)
+	if dur := c.config().Sinker.ProwJobListTimeout; dur != nil {
+		val := int64(dur.Duration.Seconds())
+		timeoutSecs = &val
+	}
+	listOpts := &ctrlruntimeclient.ListOptions{
```

Comment:
We use a cache-backed client for sinker, this is effectively a timeout for copying the objects in memory

Reply:
ACK, updated to only change plank.